service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageDocumentService.java

```diff
@@ -220,6 +216,25 @@
         return updatedC2Bundles;
     }
 
+    public List<Element<SupportingEvidenceBundle>> setDateTimeOnHearingFurtherEvidenceSupportingEvidence(
+        CaseData caseData, CaseData caseDataBefore) {
+        List<Element<SupportingEvidenceBundle>> currentSupportingDocuments
+            = caseData.getSupportingEvidenceDocumentsTemp();
+        List<Element<SupportingEvidenceBundle>> previousSupportingDocuments = List.of();
+
+        UUID selectedHearingCode = getDynamicListValueCode(caseData.getManageDocumentsHearingList(), mapper);
+
+        if (caseDataBefore.documentBundleContainsHearingId(selectedHearingCode)) {
+            for (Element<HearingFurtherEvidenceBundle> element : caseDataBefore.getHearingFurtherEvidenceDocuments()) {
+                if (element.getId().equals(selectedHearingCode)) {
+                    previousSupportingDocuments = element.getValue().getSupportingEvidenceBundle();
+                }
+            }
+        }
```

Comment:
maybe 
` List<Element<SupportingEvidenceBundle>> previousSupportingDocuments = ElementUtils.findElement(selectedHearingCode, caseDataBefore.getHearingFurtherEvidenceDocuments())
            .map(Element::getValue)
            .map(HearingFurtherEvidenceBundle::getSupportingEvidenceBundle)
            .orElse(List.of());`